document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -350,6 +350,32 @@ public boolean delete(Key key) {
     return false;
   }
 
+  @Override
+  public boolean delete(Filter filter) {
+    if (filter == null) {
+      throw new UnsupportedOperationException("Filter must be provided");
+    }
+    StringBuilder sqlBuilder = new StringBuilder("DELETE FROM ").append(collectionName);
+    Params.Builder paramsBuilder = Params.newBuilder();
+    String filters = PostgresQueryParser.parseFilter(filter, paramsBuilder);
+    LOGGER.debug("Sending query to PostgresSQL: {} : {}", collectionName, filters);
+
+    if (filters != null) {
```

Comment:
Throw exception even if the parsedFilters turns out to be `null`?
We never want a scenario to accidentally delete everything.

Reply:
I don't think it will ever be null based on parsed filter execution but no harm in adding extra checks

Reply:
Makes sense. It would ensure even if the implementation of "parseFilter()" changes tomorrow. 👍